source/agora/test/Base.d

```diff
@@ -1106,39 +1106,45 @@ public class TestAPIManager
         Assert all the nodes contain the same blocks
 
         This is a helper function to confirm all nodes have the same blocks
+        Note that the `from` and `to` are reversed to enable default value
 
         Params:
             client_idxs = client indices for the nodes to be checked
-            height = expected block height of the nodes
+            to = expected block height of the nodes
+            from = start of range for comparing the blocks
 
     ***************************************************************************/
 
-    void assertSameBlocks (Height height,
+    void assertSameBlocks (Height to, Height from = Height(0),
         string file = __FILE__, int line = __LINE__)
     {
-        assertSameBlocks(iota(GenesisValidators), height, file, line);
+        assertSameBlocks(iota(GenesisValidators), to, from, file, line);
     }
 
     /// Ditto
-    void assertSameBlocks (Idxs)(Idxs client_idxs, Height height,
-        string file = __FILE__, int line = __LINE__)
+    void assertSameBlocks (Idxs)(Idxs client_idxs, Height to,
+        Height from = Height(0), string file = __FILE__, int line = __LINE__)
     {
         static assert (isInputRange!Idxs);
+        const MaxBlocks = 1024;
+        assert(to >= from,
+            format!"[%s:%s] Please provide valid heights as params. Not %s .. %s"
+            (file, line, from, to));
 
         client_idxs.each!(idx =>
-            retryFor(Height(this.clients[idx].getBlockHeight()) == height,
+            retryFor(Height(this.clients[idx].getBlockHeight()) == to,
                 5.seconds,
                 format!"[%s:%s] Expected height %s for client #%s not %s"
-                    (file, line, height, idx,
+                    (file, line, to, idx,
                         this.clients[idx].getBlockHeight())));
 
         retryFor(client_idxs.map!(idx =>
-            this.clients[idx].getAllBlocks()).uniq().count() == 1,
+            this.clients[idx].getBlocksFrom(from, MaxBlocks)).uniq().count() == 1,
```

Comment:
Doesn't the node limit it to 10 or 20 ?

Reply:
In that case we need to do batches, which actually makes more sense as comparing more than 5 blocks at a time is not easy.

Reply:
In fact maybe better to compare a block at a time. What do you think?

Reply:
That'd make sense, but how do you plan to implement it ?

Reply:
Using range I'll check at each height. Can now revert Simple test code to also just target a height in a single call as each block will be checked and if fails will show the blocks for each node at that height.